controllers/renovate_util.go

```diff
@@ -268,27 +308,127 @@ func (u ComponentDependenciesUpdater) GetUpdateTargetsGithubApp(ctx context.Cont
+			if customRenovateConfigMapName != NamespaceWideRenovateConfigName {
+				log.Info("custom renovate config map in component annotation doesn't exist", "configMapName", customRenovateConfigMapName)
+			}
+			// no need to fail if custom renovate config map doesn't exist
+			return customRenovateOptions, nil
+		}
+		return customRenovateOptions, err
+	}
+	log.Info("will use custom renovate config map", "configMapName", customRenovateConfigMapName)
+
+	automergeOption, automergeExists := customRenovateConfigMap.Data[RenovateConfigMapAutomergeKey]
+	if automergeExists {
+		automergeValue, err := strconv.ParseBool(automergeOption)
+		if err != nil {
+			log.Error(err, "can't parse automerge option in configmap", "configMapName", customRenovateConfigMapName, "automergeValue", automergeOption)
+		} else {
+			customRenovateOptions.Automerge = automergeValue
+		}
+	}
+
+	platformAutomergeOption, platformAutomergeExists := customRenovateConfigMap.Data[RenovateConfigMapPlatformAutomergeKey]
+	if platformAutomergeExists {
+		platformAutomergeValue, err := strconv.ParseBool(platformAutomergeOption)
+		if err != nil {
+			log.Error(err, "can't parse platformAutomerge option in configmap", "configMapName", customRenovateConfigMapName, "platformAutomergeValue", platformAutomergeOption)
+		} else {
+			customRenovateOptions.PlatformAutomerge = platformAutomergeValue
+		}
+	}
+
+	ignoreTestsOption, ignoreTestsExists := customRenovateConfigMap.Data[RenovateConfigMapIgnoreTestsKey]
+	if ignoreTestsExists {
+		ignoreTestsValue, err := strconv.ParseBool(ignoreTestsOption)
+		if err != nil {
+			log.Error(err, "can't parse ignoreTests option in configmap", "configMapName", customRenovateConfigMapName, "ignoreTestsValue", ignoreTestsOption)
+		} else {
+			customRenovateOptions.IgnoreTests = ignoreTestsValue
+		}
+	}
+
+	automergeTypeOption, automergeTypeExists := customRenovateConfigMap.Data[RenovateConfigMapAutomergeTypeKey]
+	if automergeTypeExists {
+		customRenovateOptions.AutomergeType = automergeTypeOption
+	}
+
+	commitMessagePrefixOption, commitMessagePrefixExists := customRenovateConfigMap.Data[RenovateConfigMapCommitMessagePrefixKey]
+	if commitMessagePrefixExists {
+		customRenovateOptions.CommitMessagePrefix = commitMessagePrefixOption
+	}
+
+	commitMessageSuffixOption, commitMessageSuffixExists := customRenovateConfigMap.Data[RenovateConfigMapCommitMessageSuffixKey]
+	if commitMessageSuffixExists {
+		customRenovateOptions.CommitMessageSuffix = commitMessageSuffixOption
+	}
+
+	fileMatchOption, fileMatchExists := customRenovateConfigMap.Data[RenovateConfigMapFileMatchKey]
+	if fileMatchExists {
+		fileMatchParts := strings.Split(fileMatchOption, ",")
+		for i := range fileMatchParts {
+			fileMatchParts[i] = strings.TrimSpace(fileMatchParts[i])
+		}
+		customRenovateOptions.FileMatch = fileMatchParts
+	}
+
+	return customRenovateOptions, nil
+}
+
+func readComponentCustomRenovateConfigMapAnnotation(component *v1alpha1.Component) string {
```

Comment:
Please add description which says that the annotation value, which is config map with custom renovate settings, is returned.

Reply:
ok